### DISCUSSION TITLE
API for MultiViewer

### DISCUSSION THREAD

**Discussion by jtokash** [f1multiviewer/issue-tracker#57] [Feature requests] [answered]
I would love an API.  Some example use cases:

Push commands
- Move HAM video to Display 1.  Resize to 0,0 with a width of 500
- Set International video to Volume 0
- Set all videos to timecode x

Less important: Request data
- Active video window list (including name, current timecode, volume, display #, x/y/w/h)
- Driver positions
- Sector times

An API like this would take some pressure off of other feature requests, letting users get pretty custom with their setups.

Instead of an API it may be better to think about extending MultiViewer the way Obsidian does plugins- https://marcus.se.net/obsidian-plugin-docs/concepts/plugin-anatomy

**Comment by jeppeolesen** [comment, marked as answer]
Hey there, two year old issue, but still 😅 

MultiViewer now has an embedded Graph API that you can access when the app is open.

When the app is open, you can access http://localhost:10101/api/graphql in you browser of choice to view the schema. If you go to "Settings" -> "Experimental settings" there's also a couple of options to play with the API.
